test/core/image_processing/simple_kernels.cpp

```diff
@@ -0,0 +1,142 @@
+#include <boost/gil/image_processing/numeric.hpp>
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/typedefs.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+namespace gil = boost::gil;
+
+bool are_equal(gil::gray32f_view_t expected, gil::gray32f_view_t actual)
+{
+    if (expected.dimensions() != actual.dimensions())
+        return false;
+
+    for (long int y = 0; y < expected.height(); ++y)
```

Comment:
I'll be a stubborn old man about `std::ptrdiff_t` fits here better than `long int`  :)